crates/shielder-scheduler-server/src/storage/dynamo_db.rs

```diff
@@ -0,0 +1,475 @@
+use aws_sdk_dynamodb::{
+    error::SdkError,
+    operation::{describe_table::DescribeTableError, put_item::PutItemError},
+    types::{
+        AttributeDefinition, AttributeValue, GlobalSecondaryIndex, KeySchemaElement, KeyType,
+        Projection, ProjectionType, ProvisionedThroughput, ScalarAttributeType,
+    },
+    Client,
+};
+use chrono::{DateTime, Utc};
+
+use crate::storage::{RequestStatus, ScheduledRequest, StorageError, StorageInterface};
+
+const TABLE_NAME: &str = "ScheduledRequests";
+const STATUS_RELAY_INDEX: &str = "StatusRelayAfterIndex";
+
+pub struct DynamoDb {
+    client: Client,
+}
+
+impl DynamoDb {
+    pub async fn new() -> Result<Self, StorageError> {
+        let config = aws_config::load_defaults(aws_config::BehaviorVersion::latest()).await;
+        let client = Client::new(&config);
+        let db = Self { client };
+        db.create_table_if_not_exists().await?;
+        Ok(db)
+    }
+
+    async fn create_table_if_not_exists(&self) -> Result<(), StorageError> {
+        // Describe the table. If it exists return Ok
+        match self
+            .client
+            .describe_table()
+            .table_name(TABLE_NAME)
+            .send()
+            .await
+        {
+            Ok(_) => return Ok(()),
```

Comment:
Please add some INFO log line, that logs successful table creation (make sure table name is logged).